lib/timeline/component/Group.js

```diff
@@ -498,7 +496,7 @@ Group.prototype._updateSubgroupsSizes = function () {
       })
 
       me.subgroups[subgroup].start = newStart;
-      me.subgroups[subgroup].end = newEnd;
+      me.subgroups[subgroup].end = new Date(newEnd - 1) // -1 to compensate for colliding end to start subgroups;
```

Comment:
Why is the conversion to `Date` necessary here? Field `start` doesn't appear to need it, and the code works fine without it.

Also, the result is that it's possible to have the new end become `newEnd - 2`. I know this is nitpicking about milliseconds, but still....I present an alternative in the main review comment, see if that makes sense (I am a virgin with respect to `timeline`).